internal/controller/snapshot/snapshot_controller_test.go

```diff
@@ -198,6 +198,15 @@ var _ = Describe("SnapshotController", func() {
 		hasSnapshot.Spec.Application = string("im-a-ghost")
 		Expect(k8sClient.Update(ctx, hasSnapshot)).Should(Succeed())
 
+		// Use Eventually to ensure the update has been applied
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Namespace: hasSnapshot.Namespace,
+				Name:      hasSnapshot.Name,
+			}, hasSnapshot)
+			return err == nil && hasSnapshot.Spec.Application == "im-a-ghost"
+		}, time.Second*20).Should(BeTrue())
```

Comment:
would it be better if we set the interval poll argument
```suggestion
		}, time.Second*20, time.Second*2).Should(BeTrue())
```

Reply:
I can do that if you wish. I tested it extensively in a local container with this setup at the point where the test was failing (with the old code) and its not hitting that time out at the low point of resources. If setting a poll interval is best practice then no problem to do that. LMK.